modules/navier_stokes/include/physics/PNSFVSolidHeatTransferPhysics.h

```diff
@@ -0,0 +1,73 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "HeatConductionFV.h"
+#include "NS.h"
+
+/**
+ * Creates all the objects needed to solve the porous media solid energy equation
+ */
+class PNSFVSolidHeatTransferPhysics final : public HeatConductionFV
+{
+public:
+  static InputParameters validParams();
+
+  PNSFVSolidHeatTransferPhysics(const InputParameters & parameters);
+
+protected:
+private:
+  void addNonlinearVariables() override;
+  void addFVKernels() override;
+  void addMaterials() override;
+
+  // Note that we inherit:
+  // - addInitialConditions from HeatConductionPhysics
+  // - addPreconditioning from HeatConductionPhysics
+  // - addFVBCs from HeatConductionFV
+
+  InputParameters getAdditionalRMParams() const override;
+
+  /**
+   * Functions adding kernels for the solid energy equation
+   */
+  void addPINSSolidEnergyTimeKernels();
+  void addPINSSolidEnergyHeatConductionKernels();
+  void addPINSSolidEnergyAmbientConvection();
+  void addPINSSolidEnergyExternalHeatSource();
+
+  // TODO Add interfaces
```

Comment:
Add more information in this comment if you want to keep it around

Reply:
removed